### PR TITLE
feat(search): Add capability to run time range query

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

This adds support for being able to run queries such that you can select a specific interval of time

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Unit tests

### CODE REVIEW

File: pkg/search/postgres/query/time_query.go [hunk 1]
```diff
@@ -62,6 +76,29 @@ func timeQueryEntry(ctx *queryAndFieldContext, prefix, formattedTime string) *Qu
 	}, nil)
 }
 
+func parseTimeRange(value string) (from, to time.Time, ok bool, err error) {
```
Comment (dashrews78) [inline pkg/search/postgres/query/time_query.go:79]: nit:  A comment here or in the calling function calling out that it is expected query builder converted the times to milliseconds would have helped me when reviewing this.

**Review by dashrews78 — APPROVED** [review]

File: pkg/search/postgres/query/time_query.go (hunk 1 above)
Comment (connorgorman) [reply pkg/search/postgres/query/time_query.go:79]: Yeah let me add a comment

File: pkg/search/postgres/query/time_query.go [hunk 2]
```diff
@@ -18,6 +21,17 @@ func newTimeQuery(ctx *queryAndFieldContext) (*QueryEntry, error) {
 	if len(ctx.queryModifiers) > 0 {
 		return nil, errors.New("modifiers not supported for time query")
 	}
+	from, to, ok, err := parseTimeRange(ctx.value)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing time range")
+	}
+	if ok {
+		return qeWithSelectFieldIfNeeded(ctx, &WhereClause{
+			Query:  fmt.Sprintf("%s >= $$ and %s < $$", ctx.qualifiedColumnName, ctx.qualifiedColumnName),
```
Comment (md2119) [inline pkg/search/postgres/query/time_query.go:30]: Can we take this opportunity to add support for >= or <= as well?

**Review by md2119 — CHANGES_REQUESTED** [review]
Since you are already adding support for `>= and <=`, I feel we should take this opportunity to add support for queries with only `>=` and `<=`.

**Comment by connorgorman** [comment]
> Since you are already adding support for >= and <=, I feel we should take this opportunity to add support for queries with only >= and <=

Supporting this will really make the language more complicated. Also, the current code makes this possible today, you just need to add one ms. Without a use case, I'm not sure it's worth it

**Comment by connorgorman** [comment]
To follow up on my previous comment, I would rather not add double ended inclusive now and instead refactor the queries so they don't come into postgres as unstructured and instead would come in as a TimeRangeQuery where then we can customize it much more

**Review by md2119 — APPROVED** [review]
There are quite a few possibilities such as searching for image scans earlier than a x, violations before x ts. It's worth noting that due to the absence of dedicated time-range query support, we've been constructing queries using `AddStrings` method.

I wanted to share this notion since you're already tinkering in this area. Of course, feel free to explore it only if you find it aligns with your current motivation and objectives.

**Pull request merged**